### ISSUE TITLE
Crash on get-config with empty filter

### ISSUE THREAD

**Issue by cmoberg** [FlowForwarding/LINC-Switch#70]
During some interoperability testing between a client implementation and the LINC NETCONF server, we found a way to crash the server with an edit-config operation with an empty subtree filter.

We managed to provoke a crash by doing a get-config using an empty filter statement. The first RPC content that triggered it is this:

```
 <get-config>
   <source>
     <running/>
   </source>
   <filter/>
 </get-config>
</rpc>
```

We then removed the filter element:

```
 <get-config>
   <source>
     <running/>
   </source>
 </get-config>
</rpc>
```

...and it didn't crash.
We then added a non-empty filter:

```
<get-config>
   <source>
     <running/>
   </source>
   <filter type="subtree">
     <top xmlns="urn:onf:of111:config:yang"/>
   </filter>
 </get-config>
```

...and it didn't crash.
We then added another version of empty filter:

```
<get-config>
   <source>
     <running/>
   </source>
   <filter type="subtree">
   <filter></filter>
 </get-config>
```

...and it crashed.

So, in summary. It is likely that an empty filter element is the reason for the behaviour.
